src/plugins/gui/providers/kadoProvider.ts

```diff
@@ -561,17 +559,31 @@ export const kadoProvider: FiatProviderFactory = {
 
         const { amount: minAmount, unit: minUnit } = quote.data.quote.minValue
         const { amount: maxAmount, unit: maxUnit } = quote.data.quote.maxValue
+        const { baseAmount } = quote.data.quote
 
         let fiatAmount: string
         let cryptoAmount: string
+        const { totalFee } = quote.data.quote
+        const { amount, unitCount } = quote.data.quote.receive
         if (direction === 'buy') {
-          const { unitCount } = quote.data.quote.receive
-          cryptoAmount = unitCount.toString()
-          fiatAmount = exchangeAmount
+          if (amountType === 'fiat') {
+            cryptoAmount = unitCount.toString()
+            fiatAmount = exchangeAmount
+          } else {
+            // For some reason, the totalFee is not included in the fiatAmount on buy reverse quotes
+            fiatAmount = (amount + totalFee.amount).toString()
+            cryptoAmount = exchangeAmount
+          }
         } else {
-          const { amount } = quote.data.quote.receive
-          cryptoAmount = exchangeAmount
-          fiatAmount = amount.toString()
+          if (amountType === 'crypto') {
+            cryptoAmount = exchangeAmount
+            fiatAmount = amount.toString()
+          } else {
+            fiatAmount = exchangeAmount
+            // The unitCount is 0 for sell reverse quotes but seems to be in
+            // the baseAmount.amount so use that
+            cryptoAmount = baseAmount.amount.toString()
+          }
```

Comment:
```ts
const quote = asQuoteResponse(result).data
// ...
cryptoAmount = quote.baseAmount.amount.toString()
```

Removing the destructuring will keep the context of where `baseAmount` came from. 

It would also avoid the need to rename things like `minAmount` and `minUnit` when you can just use `quote.minValue.amount` and `quote.minValue.unit`. It's more to read where it's used, but it avoids needing to read paragraphs of destructuring and renaming.

Reply:
I agree. I left the `minAmount` and `minUnit` as is to reduce modification of the existing code. For the baseAmount, the destructuring seems harmless as there is no ambiguity as to where it came from.